test/gtest/ucp/test_ucp_proto_mock.cc

```diff
@@ -891,101 +892,165 @@ class test_ucp_proto_mock_rcx_twins : public test_ucp_proto_mock {
+};
+
+void test_ucp_proto_mock_rcx_twins_put::check_config(
+    const proto_select_data_vec_t &data_vec)
+{
+    uint8_t dummy;
+    ucp_mem_map_params_t mem_map_params;
+    mem_map_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                                UCP_MEM_MAP_PARAM_FIELD_LENGTH;
+    mem_map_params.address    = &dummy;
+    mem_map_params.length     = sizeof(dummy);
+
+    ucp_mem_h mem;
+    ASSERT_UCS_OK(ucp_mem_map(receiver().ucph(), &mem_map_params, &mem));
+
+    void *rkey_buffer;
+    size_t rkey_buffer_size;
+    ASSERT_UCS_OK(ucp_rkey_pack(receiver().ucph(), mem, &rkey_buffer,
+                                &rkey_buffer_size));
+
+    ucp_rkey_h rkey;
+    ASSERT_UCS_OK(ucp_ep_rkey_unpack(sender().ep(), rkey_buffer, &rkey));
+
+    ucp_rkey_destroy(rkey);
+    ucp_rkey_buffer_release(rkey_buffer);
+    ASSERT_UCS_OK(ucp_mem_unmap(receiver().ucph(), mem));
+
     ucp_proto_select_key_t key = any_key();
-    key.param.op_id_flags      = UCP_OP_ID_AM_SEND;
+    key.param.op_id_flags      = UCP_OP_ID_PUT;
     key.param.op_attr          = 0;
 
-    check_ep_config(sender(),
-                    {
-                            {0, 200, "short", "rc_mlx5/mock_0:1/path0"},
-                            {201, 404, "copy-in", "rc_mlx5/mock_0:1/path0"},
-                            {405, 8246, "zero-copy", "rc_mlx5/mock_0:1/path0"},
-                            {8247, 18540, "multi-frag zero-copy",
-                             "rc_mlx5/mock_0:1/path0"},
-                            // SINGLE_NET_DEVICE=y, NODE_LOCAL_ID=2
-                            // Use two paths of the first network device in
-                            // 50/50 ratio
-                            {18541, INF,
-                             "rendezvous zero-copy read from remote",
-                             "rc_mlx5/mock_0:1 50% on path0 and 50% on path1"},
-                    },
-                    key);
+    check_rkey_config(sender(), data_vec, key);
+}
+
+UCS_TEST_P(test_ucp_proto_mock_rcx_twins_put, use_single_net_device_rank_0,
+           "SINGLE_NET_DEVICE=y", "NODE_LOCAL_ID=0")
+{
+    check_config({
+                        {0, 2048, "short", "rc_mlx5/mock_0:1/path0"},
+                        {2049, INF, "zero-copy", "rc_mlx5/mock_0:1/path0"}
+                });
+}
+
+UCS_TEST_P(test_ucp_proto_mock_rcx_twins_put, use_single_net_device_rank_1,
+           "SINGLE_NET_DEVICE=y", "NODE_LOCAL_ID=1")
+{
+    check_config({
+                        {0, 2048, "short", "rc_mlx5/mock_0:1/path0"},
+                        {2049, INF, "zero-copy", "rc_mlx5/mock_1:1/path0"}
+                });
+}
+
+UCS_TEST_P(test_ucp_proto_mock_rcx_twins_put, use_single_net_device_rank_2,
+           "SINGLE_NET_DEVICE=y", "NODE_LOCAL_ID=2")
+{
+    check_config({
+                        {0, 2048, "short", "rc_mlx5/mock_0:1/path0"},
+                        {2049, INF, "zero-copy", "rc_mlx5/mock_0:1/path0"}
```

Comment:
why not mock_2:1?

Reply:
mock_2:1 has lower bw.

Reply:
mabye a better stress test would be to make mock0 and mock2 have the same BW, while mock1 would have lower bw?